diff --git a/src/libasr/codegen/llvm_utils.h b/src/libasr/codegen/llvm_utils.h
index c42f81a79..29183ddce 100644
--- a/src/libasr/codegen/llvm_utils.h
+++ b/src/libasr/codegen/llvm_utils.h
@@ -1214,7 +1214,7 @@ class LLVMFinalize {
             if(is_class_type){
                 auto const class_type_llvm = llvm_utils_->getClassType(struct_sym);
                 auto const consecutive_field_ptr = llvm_utils_->CreateGEP2(class_type_llvm, data_ptr, 1);
-                auto const consecutive_field_type = consecutive_field_ptr->getType()->getPointerElementType();
+                auto const consecutive_field_type = class_type_llvm->getStructElementType(1);
                 auto const allocated_cosecutive_structs = builder_->CreateLoad(consecutive_field_type, consecutive_field_ptr);
                 llvm_utils_->lfortran_free(allocated_cosecutive_structs);
                 // deallocate class wrapper 
